service_capacity_modeling/models/org/netflix/time_series.py

```diff
@@ -0,0 +1,152 @@
+from typing import Any
+from typing import Dict
+from typing import Optional
+from typing import Sequence
+from typing import Tuple
+
+from .stateless_java import nflx_java_app_capacity_model
+from service_capacity_modeling.interface import AccessPattern
+from service_capacity_modeling.interface import CapacityDesires
+from service_capacity_modeling.interface import CapacityPlan
+from service_capacity_modeling.interface import DataShape
+from service_capacity_modeling.interface import Drive
+from service_capacity_modeling.interface import FixedInterval
+from service_capacity_modeling.interface import Instance
+from service_capacity_modeling.interface import Interval
+from service_capacity_modeling.interface import QueryPattern
+from service_capacity_modeling.interface import RegionContext
+from service_capacity_modeling.models import CapacityModel
+
+
+class NflxTimeSeriesCapacityModel(CapacityModel):
+    @staticmethod
+    def capacity_plan(
+        instance: Instance,
+        drive: Drive,
+        context: RegionContext,
+        desires: CapacityDesires,
+        extra_model_arguments: Dict[str, Any],
+    ) -> Optional[CapacityPlan]:
+        # TimeSeries wants 20GiB root volumes
+        extra_model_arguments.setdefault("root_disk_gib", 20)
+
+        ts_app = nflx_java_app_capacity_model.capacity_plan(
+            instance=instance,
+            drive=drive,
+            context=context,
+            desires=desires,
+            extra_model_arguments=extra_model_arguments,
+        )
+        if ts_app is None:
+            return None
+
+        for cluster in ts_app.candidate_clusters.regional:
+            cluster.cluster_type = "dgwts"
+        return ts_app
+
+    @staticmethod
+    def description():
+        return "Netflix Streaming TimeSeries Model"
+
+    @staticmethod
+    def extra_model_arguments() -> Sequence[Tuple[str, str, str]]:
+        return nflx_java_app_capacity_model.extra_model_arguments()
+
+    @staticmethod
+    def compose_with(
+        user_desires: CapacityDesires, extra_model_arguments: Dict[str, Any]
+    ) -> Tuple[str, ...]:
+        # In the future depending on the user desire we might need EVCache
```

Comment:
nit: depending on the extra model arguments we'll need to add Elasticsearch

Reply:
How do you envision the arguments to be? Would that be an enum that ties to a model name or the model name itself? Should we add that in the next iteration when we are adding support for ES?

Reply:
The extra model arguments come from the free-form parameters. So I imagine that in the UI we'd have "Require best-effort cross event search?" and that would set a boolean flag "search=True" or some such in extra_model_arguments.

Then this function can return both elastic and C* when we are composing them.